account_controllers/controllers.js

```diff
@@ -176,20 +223,111 @@ module.exports = {
 
     editOperationHours: async (req, res) => {
       try {
-        const { id } = req.params;
         const { operation_hours } = req.body;
-        const { error } = await supabase
+        const { id } = req.params;
+        const [startHour, endHour] = operation_hours.split('-').map(Number);
+
+        if (
+          startHour < 0 ||
+          startHour > 24 ||
+          endHour < 0 ||
+          endHour > 24 ||
+          startHour >= endHour
+        ) {
+          return res.status(400).send('Invalid hours');
+        }
+
+        const { data: garagesData, error: garagesError } = await supabase
           .from('garages')
-          .update({ operation_hours })
+          .select('operation_hours')
           .eq('user_id', id)
           .single();
-        if (error) {
-          console.error('Error updating operation hours:', error);
+
+        if (garagesError) {
+          console.error(
+            'Error fetching current operation hours:',
+            garagesError
+          );
+          return res.sendStatus(500);
+        }
+
+        const currentOperationHours = garagesData
+          ? garagesData.operation_hours
+          : null;
+
+        if (!currentOperationHours) {
+          return res.status(400).send('Current operation hours not found');
+        }
+
+        const [currentStartHour, currentEndHour] = currentOperationHours
+          .split('-')
+          .map(Number);
+
+        const today = DateTime.local().startOf('day');
+        const next7Days = [...Array(7)].map((_, index) =>
+          today.plus({ days: index })
+        );
+        const formattedNext7Days = next7Days.map((day) =>
```

Comment:
Future refactor? - I wonder if these two map functions can be chained?

Reply:
Just refactor today, next7Days and formattedNext7Days into one function.